src/overlay005/vs_seeker.c

```diff
@@ -0,0 +1,903 @@
+            VsSeekerSystem_StartAnimation(vsSeeker, vsSeeker->trainers[i], sVsSeekerAnimSingleExclamationMark);
+            anyAvailable = TRUE;
+        } else {
+            if (((LCRNG_Next() % 100) < VS_SEEKER_REMATCH_CHANCE) && 
+                (VsSeeker_IsTrainerDoingRematchAnimation(vsSeeker->trainers[i]) == FALSE)) {
+                VsSeeker_SetTrainerMoveCode(vsSeeker->trainers[i], 0x31);
+                VsSeekerSystem_StartAnimation(vsSeeker, vsSeeker->trainers[i], sVsSeekerAnimDoubleExclamationMark);
+
+                MapObject *secondTrainer = VsSeeker_GetSecondDoubleBattleTrainer(vsSeeker->fieldSystem, vsSeeker->trainers[i], VS_SEEKER_2V2_TRAINER_SEARCH_MODE_REMATCH_ANIM_CHECK);
+
+                if (secondTrainer != NULL) {
+                    VsSeeker_SetTrainerMoveCode(secondTrainer, 0x31);
+                    VsSeekerSystem_StartAnimation(vsSeeker, secondTrainer, sVsSeekerAnimDoubleExclamationMark);
+                }
+
+                anyAvailable = TRUE;
+                Events_SetVsSeekerUsedFlag(events);
+            } else {
+                (void)0;
+            }
+        }
+    }
+
+    return anyAvailable;
+}
+
+static u16 VsSeeker_GetTrainerIDFromMapObject(MapObject *trainerObj)
+{
+    u32 eventID = MapObject_GetEventID(trainerObj);
+    return Script_GetTrainerIDFromEventID(eventID);
+}
+
+u16 VsSeeker_GetRematchTrainerID(FieldSystem *fieldSystem, MapObject *trainerObj, u16 trainerID)
+{
+    if (VsSeeker_IsTrainerDoingRematchAnimation(trainerObj) == FALSE) {
+        return 0;
+    }
+
+    u16 index = VsSeeker_GetRematchDataIndexForTrainer(fieldSystem, trainerID);
+    if (index == VS_SEEKER_REMATCH_DATA_INDEX_NONE) {
+        return 0;
+    }
+
+    u16 level = VsSeeker_GetCurrentLevelForRematchData(fieldSystem, index);
+    level = VsSeeker_AdjustRematchLevel(fieldSystem, index, level);
+    return VsSeeker_GetTrainerIDForRematchLevel(index, level);
+}
+
+static u16 VsSeeker_GetRematchDataIndexForTrainer(FieldSystem *fieldSystem, u16 trainerID)
+{
+    for (int i = 0; i < (NELEMS(gVsSeekerRematchData)); i++) {
+        if (gVsSeekerRematchData[i].trainerIDs[0] != trainerID) {
+            continue;
+        }
+
+        return i;
+    }
+
+    return VS_SEEKER_REMATCH_DATA_INDEX_NONE;
+}
+
+static u16 VsSeeker_GetCurrentLevelForRematchData(FieldSystem *fieldSystem, u16 rematchDataIndex)
+{
+    int level;
+    const VsSeekerRematchData *rematchData = gVsSeekerRematchData;
+
+    for (level = 1; level < VS_SEEKER_MAX_REMATCHES; level++) {
+        if (rematchData[rematchDataIndex].trainerIDs[level] == VS_SEEKER_REMATCH_DATA_END) {
+            return level - 1;
+        }
+
+        if (rematchData[rematchDataIndex].trainerIDs[level] != VS_SEEKER_REMATCH_DATA_DUMMY) {
+            if (Script_HasBeatenTrainer(fieldSystem, rematchData[rematchDataIndex].trainerIDs[level]) == FALSE) {
+                return level;
+            }
+        }
```

Comment:
```suggestion
        if (rematchData[rematchDataIndex].trainerIDs[level] != VS_SEEKER_REMATCH_DATA_DUMMY
            && Script_HasBeatenTrainer(fieldSystem, rematchData[rematchDataIndex].trainerIDs[level]) == FALSE) {
            return level;
        }
```

Or, even better, if it matches:

```suggestion
        u16 trainerID = rematchData[rematchDataIndex].trainerIDs[level];
        if (trainerID != VS_SEEKER_REMATCH_DATA_DUMMY
            && Script_HasBeatenTrainer(fieldSystem, trainerID) == FALSE) {
            return level;
        }
```